src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/DatabaseTreeNode.cs

```diff
@@ -102,5 +102,12 @@ private bool IsDWGen3(Database db)
                 && db.DatabaseEngineEdition == DatabaseEngineEdition.SqlDataWarehouse
                 && db.ServerVersion.Major == 12;
         }
+
+        private bool IsDataMart(Database db){
+            // Datamarts have a DatabaseEngineEdition of 1000. We have to cast it to int as DatabaseEngineEdition enum does not have a value for Datamarts.
```

Comment:
Time to update the enum definition?

Reply:
Yup. This probably has to be updated in smo

Reply:
@shueybubbles FYI

Reply:
SMO should probably report this as "Unknown" instead of accepting random values